infra/fargate/main.tf

```diff
@@ -0,0 +1,210 @@
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": [
+          "ecs-tasks.amazonaws.com"
+        ]
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "query_runner" {
+  name = "dss-monitor-query-runner-${var.DSS_DEPLOYMENT_STAGE}"
+  role = "${aws_iam_role.query_runner.id}"
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+            {
+      "Effect": "Allow",
+      "Action": "secretsmanager:Get*",
+      "Resource": "arn:aws:secretsmanager:*:${var.AWS_ACCOUNT_ID}:secret:${var.DSS_SECRETS_STORE}/*"
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "tag:GetTagKeys",
+        "tag:GetResources",
+        "tag:GetTagValues",
+        "cloudwatch:*"
+      ],
+      "Resource": "*"
+    }
+  ]
+}
+EOF
+
+}
+
+
+resource "aws_ecs_task_definition" "monitor" {
+  family = "dss-monitor-${var.DSS_DEPLOYMENT_STAGE}"
+  execution_role_arn = "${aws_iam_role.task_executor.arn}"
+  task_role_arn = "${aws_iam_role.query_runner.arn}"
+  requires_compatibilities = ["FARGATE"]
+  network_mode = "awsvpc"
+  cpu = "256"
+  memory = "512"
+  # TODO add logging below.
+  container_definitions = <<DEFINITION
+[
+  {
+    "family": "dss-monitor",
+    "name": "dss-monitor-lambda",
+    "image": "humancellatlas/dss-monitor-image",
+    "cpu": 256,
+    "memory": 512,
+    "essential": true,
+    "logConfiguration": {
+        "logDriver": "awslogs",
+        "options": {
+          "awslogs-group": "${aws_cloudwatch_log_group.query_runner.name}",
+          "awslogs-region": "us-east-1",
+          "awslogs-stream-prefix": "ecs"
+        }
+    },
+    "portMappings": [
+      {
+        "containerPort": 80,
+        "hostPort": 80
+      }
+    ]
+  }
+]
+DEFINITION
+  # tags = "${local.common_tags}"
+}
+
+
+resource "aws_cloudwatch_log_group" "query_runner" {
+  name              = "/aws/service/dss-monitor-query-runner-${var.DSS_DEPLOYMENT_STAGE}"
+  retention_in_days = 1827
+}
+
+
+resource "aws_vpc" "dss_fargate" {
+  cidr_block = "10.0.0.0/16"
+  enable_dns_support = true
+  enable_dns_hostnames = true
+  tags = "${local.common_tags}"
+}
+
+# These subnets need to be associated with a route table providing
+# internet access.
+resource "aws_subnet" "dss_monitor" {
+  count 		    = "${length(local.availability_zones)}"
+  vpc_id 			= "${aws_vpc.dss_fargate.id}"
+  availability_zone = "${local.availability_zones[count.index]}"
+  cidr_block = "${cidrsubnet("10.0.0.0/16", "4", count.index)}"
+}
+
+resource "aws_internet_gateway" "gw" {
+  vpc_id = "${aws_vpc.dss_fargate.id}"
+}
+
+# Route the public subnet trafic through the IGW
+resource "aws_route" "internet_access" {
+  route_table_id         = "${aws_vpc.dss_fargate.main_route_table_id}"
+  destination_cidr_block = "0.0.0.0/0"
+  gateway_id             = "${aws_internet_gateway.gw.id}"
+}
+
+
+
+resource "aws_ecs_cluster" "dss_monitor" {
```

Comment:
Let's use the default cluster for now. We can add this back in the future if needed.
```
data "aws_ecs_cluster" "default" {
  cluster_name = "default"
}
```